functions/send_encounters/main.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import os
+import secrets
+from datetime import datetime
+from typing import Optional
+
+from flask import jsonify
+from google.cloud import bigquery, datastore
+from google.cloud.datastore import Entity
+
+BQ_TABLE_ID = f"{os.environ['GCP_PROJECT']}.{os.environ['BQ_DATASET']}.{os.environ['BQ_TABLE']}"
+
+datastore_client = datastore.Client()
+
+
+def send_encounters(request):
+    if not request.is_json:
+        return jsonify(
+            {'status': 'failed',
+             'message': 'invalid data'
+             }
+        ), 422
+
+    request_data = request.get_json()
+
+    for key in ["user_id", "platform", "os_version", "device_type", "app_version", "lang", "encounters"]:
```

Comment:
You have to be careful using `user_id` from the request. It would be difficult to generate the existing `user_id`, but it is really easy to leak this data.